src/Parser/src/Parser.cpp

```diff
@@ -5,38 +5,150 @@
 */
 ////////////////////////////////////////////////////////////////////////////////
 
+#include <algorithm>
 #include <iostream>
+#include <memory>
 #include <regex>
 #include <string>
+#include <thread>
+#include <utility>
+#include <vector>
 
 #include "CalculatorMessages.h"
 #include "Parser.h"
 
 
 namespace calculator
 {
-    Expression Parser::parseFullEquation(const std::string &fullEquation)
+    std::pair <std::shared_ptr<std::vector<ExpressionUnit>>,bool> Parser::createVector(const std::string &fullEquation)
+    { 
+        editedEquation = fullEquation;
+        std::shared_ptr<std::vector<ExpressionUnit>> equationVector;
+        equationVector = std::make_shared<std::vector<ExpressionUnit>>();
+        ExpressionUnit currentUnit;
+        bool checkingNumber = true;
+        bool validVector = true;
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(300));

```

Comment:
Add a comment here explaining why you are sleeping here... Bonus points: add a preprocessor command to disable this for release